packages/webpack-plugin/src/compiler.ts

```diff
@@ -64,9 +64,16 @@ function compileVanillaSource(
       loader._compiler.webpack && loader._compiler.webpack.version,
     );
     const compat = createCompat(isWebpack5);
-    // Child compiler will compile vanilla-extract files to be evaled during compilation
-    const outputOptions = { filename: loader.resourcePath };
 
+    // Escape webpack template strings in output files so they don't get replaced
+    // Non-standard escape syntax, see the docs https://webpack.js.org/configuration/output/#template-strings
+    const outputOptions = {
+      filename: loader.resourcePath
+        .replaceAll('[', '[\\')
+        .replaceAll(']', '\\]'),
```

Comment:
Thinking about this some more, this would break a path that contains only a `[` or `]`, or some combination of non-matching square brackets. _OR_ if the path contains something similar to [a webpack template string](https://webpack.js.org/configuration/output/#template-strings) but isn't, e.g. `[myThing]`.

Maybe we could do something like `/\[(ext|id|...)\]/` for all the possible strings, which would be much more precise. Would be nice if there was a list of all template strings.

Reply:
I think we can check if the resourcePath matches `/\[.*\]/` and then do the replacements. If it hasn't been an issue so far, it likely won't be an issue in the future.